src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -190,9 +190,9 @@ private static SyntaxNode ConcatinateTextToTextNode(SyntaxGenerator generator, S
                 return false;
             }
 
-            var method = semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol as IMethodSymbol;
-            return method?.MethodKind == MethodKind.BuiltinOperator &&
-                   method.ContainingType.SpecialType == SpecialType.System_String &&
+            return semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol is IMethodSymbol method &&
+                   method.MethodKind == MethodKind.BuiltinOperator &&
+                   method.ContainingType?.SpecialType == SpecialType.System_String &&
```

Comment:
@sharwell @jasonmalinowski This is rather interesting.  We have a MethodSymbol without a null ContainingType.  I'm pretty certain there will be a lot of codepaths in the IDE that will not be resilient to this sort of thing.  The presumption was that fields/props/events/methods would all be contained in types (possibly the implicit script class for scripts).

Reply:
Ouch, that's interesting indeed. 😦 @jaredpar or maybe @jcouv do we know more when this can happen?

Reply:
I'll take a look, that does seem wrong.
What's the expression used in `semanticModel.GetSymbolInfo` in this example?

Filed https://github.com/dotnet/roslyn/issues/25377

Reply:
it's a binary + expression involving dynamic

Reply:
Perhaps the compiler synthesizes methods for dynamic invocations?  But then doesn't have an INamedTypeSymbol to place them on (because IDynamicTypeSymbol isn't an INamedTypeSymbol). (Just spitballing).

However, for this case, it would be nice if we could synthesize a type.  Similar to how top level members in script code belong to an implicit named type representing the script class.